core-plugins/src/main/java/io/cdap/plugin/batch/sink/TableSink.java

```diff
@@ -73,13 +73,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     Schema outputSchema =
       SchemaValidator.validateOutputSchemaAndInputSchemaIfPresent(tableSinkConfig.getSchemaStr(),
                                                                   tableSinkConfig.getRowField(), pipelineConfigurer);
-    if ((outputSchema != null) && (outputSchema.getFields().size() == 1)) {
-      String fieldName = outputSchema.getFields().get(0).getName();
-      if (fieldName.equals(tableSinkConfig.getRowField())) {
-        throw new IllegalArgumentException(
-          String.format("Output schema should have columns other than rowkey."));
+    if (outputSchema != null) {
+      if (outputSchema.getField(tableSinkConfig.getRowField()) == null) {
+        throw new IllegalArgumentException("Output schema should contain the rowkey column.");
+      } else if (outputSchema.getFields().size() == 1) {
```

Comment:
this should not be in an else branch. And if the size is 1, then the next check is not needed, because you know that the row field is one of the fields, that is, the only field.

Reply:
As in multiple if statements? Updated, let me know if this is what you mean.

Is there a Java best practices guide that the repo adheres to?